packages/core/admin/ee/server/controllers/content-api/workflows/stages/index.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+const { getService } = require('../../../../utils');
+const { validateUpdateStages, validateStage } = require('../../../../validation/review-workflows');
+
+module.exports = {
+  /**
+   * Get one stage
+   * @param {import('koa').BaseContext} ctx - koa context
+   */
+  async findById(ctx) {
+    const { id, workflow_id: workflowId } = ctx.params;
+    const { populate } = ctx.query;
+    const stagesService = getService('stages');
+
+    const data = await stagesService.findById(id, {
+      workflowId,
+      populate,
+    });
+
+    ctx.body = {
+      data,
+    };
+  },
+
+  /**
+   * Replace all stages in a workflow
+   * @param {import('koa').BaseContext} ctx - koa context
+   *
+   */
+  async replace(ctx) {
+    const { workflow_id: workflowId } = ctx.params;
+    const stagesService = getService('stages');
+    const {
+      body: { data: stages },
+    } = ctx.request;
+
+    const stagesValidated = await validateUpdateStages(stages);
+
+    const data = await stagesService.replaceWorkflowStages(workflowId, stagesValidated, {
+      populate: ['stages'],
+    });
+
+    ctx.body = { data };
```

Comment:
We need to sanitize the input and output here, and we need to do so in the admin controllers too.
Right now it's not an issue because workflows are not created by any user, but once we can create multiple workflows you would be able to populate user data.

Reply:
True! Thanks for the catch, I forgot about sanitization.